### DISCUSSION TITLE
如何获取CF的其他地区ip

### DISCUSSION THREAD

**Discussion by blog2020** [XIU2/CloudflareSpeedTest#471] [提问]
CloudflareST.exe -url https://pencilfiles.annonymus.cf/cloudflarest-200mb.rar  -httping -cfcolo SIN  -sl 3 -tl 300 -dn 15 -f ip.txt -o result-nrt.csv

首先感谢开发者的贡献。我想请问一下，我用上面的命令执行，显示可用ip是0，我换成了香港HKG显示有1个可用ip，但延迟很高，下载速度为0，是不是说免费用户没法获取到非美国的IP？？我查了一下2年前开发者说免费用户无法使用，2年后还是这样吗？？谢谢。

> Cloudflare 亚洲有很多节点（根据官网所写：新加坡、日本、香港、韩国、泰国、印度等等），但是绝大部分都只提供给付费客户，普通用户现在能直接用的只有香港节点，但是这个香港节点与国内内地线路一般。
电信联通访问香港节点会绕全球，只有移动直连，移动用户测速什么都不用干，测速结果最前面的一定都是延迟最低的香港节点。

> 另外，虽然移动直连，但不代表一定速度快，我前面也说了香港节点与国内内地之间线路本身就一般，主要就是 Cloudflare 无法直接进入中国市场（只能靠和京东等公司间接合作），因此为了省钱就没有针对国内内地线路优化，所以默认分配的路由都是渣渣路线，所以才会促使我写出这个软件。。。

> 以前还有个大家能自选使用的日本节点（虽然是渣渣 NTT 线路，低峰期 70ms 高峰期爆炸），但是因为用的人太多了，今年初就被 Cloudflare 限制了，无法自选指向免费套餐的域名访问了。

**Comment by XIU2** [comment]
准确的来说，亚洲节点本身就比较少，而且大都限制免费套餐的域名指向使用，剩下的一些可用的亚洲节点，Cloudflare 压根不分配给国内用户（移动倒是可以找到不少香港节点）。

你要知道 Cloudflare 的 IP 大部分都是 Anycast IP，也就是动态路由的，每个 IP 的路线和终点都不是固定的，不同地区的人访问同一个 IP，分配到的节点都是不一样的。
你在国内用你的参数跑 CloudflareST 是这个结果，但是如果你在新加坡的服务器上面扫，那就是另外一个结果了。

因此不是你找不到 新加坡 节点，而是 Cloudflare 不给你用。

**Reply by blog2020** [reply]
谢谢大佬！另外我想请教一下大佬，我有个nat机器是新加坡的，套了CF后，延迟有360+，CF不给免费用户提供新加坡ip，是否说我除了换成移动的宽带，或者用其他人提供的反代cf的新加坡节点，就没有其他办法降低延迟了？？谢谢。

**Reply by XIU2** [reply]
啥意思，你是给 新加坡 的 NAT 服务器**代理**套上 Cloudflare 吗？
这种情况是：
```
你 <=> Cloudflare CDN <=> 新加坡NAT服务器 <=> 目标网站
```
你指的 360+ ms 延迟是 Cloudflare CDN IP 的延迟，还是代理软件等方式测速得来的延迟？
如果是前者，那么就只是 `你 <=> Cloudflare CDN` 环节的延迟，而如果是后者，那么就是整个流程加起来的延迟。

**Reply by blog2020** [reply]
谢谢大佬回复，我现在情况是这样的，新加坡nat的机器ipv4是被墙的，所以要套cf，同时本地没有ipv6网络所以测不了直连的延迟，套了cf之后我在v2rayN上测速服务器真连接延迟，总是不准确，一会100多，一会又500或者700，所以我就浏览器打开fast.com和speedtest.net开全局测速，延迟都是显示360+，晚上偶尔会到500+，然后我用美西洛杉矶的机器同样套了cf的测速，延迟170左右，两台机器都是用的同一个优选ip，新加坡的nat机器延迟高，看视频拖动的时候很明显的卡顿，美西的就不会这样。这种情况还有降低新加坡nat的延迟办法吗？

**Reply by XIU2** [reply]
我印象中这些代理客户端测试延迟的原理就是，通过代理访问某个网站，然后计算全程花费的时间，如果你的代理不稳定的话，那么自然多次测速结果会波动很大。

你这种情况，要么是 新加坡 NAT 自身的网络不行，要么就是其与 Cloudflare CDN 之间的网络不行，你优选的 IP 是哪个地区的？距离新加坡太远么？
如果是 新加坡 NAT 自身网络问题，那么无解。
如果是 新加坡 NAT 与 Cloudflare CDN 之间的网络问题，那你可以尝试继续优选 IP。

除此之外，没啥其他办法了估计。

另外，你那边没有 IPv6 是你的路由器不支持，还是运营商没提供，如果运营商提供了的话，建议你最好搞个 IPv6，这对于玩代理方面还是有很大帮助的。

**Reply by blog2020** [reply]
谢谢大佬，我用的优选ip都是项目里自带的那些ip，选出来的可能是美国的ip，访问路径可能是这样：我---cdn美国---新加坡---fast.com服务器，可能是这样所以延迟高，但我不敢确定。。另外请教一下大佬，我在网上找了一些其他大佬提供的反代CF的ip，我这边把ip放入v2rayN能正常用，但把这些ip放入到CloudflareSpeedTest里测速，就是不成功，延迟和发包丢包能正常显示，速度那全是0，这个大佬能解决吗？谢谢


用的命令：CloudflareST.exe -url https://pencilfiles.rosmontics.tk/cloudflarest-200mb.rar -f worker-ip.txt -o result-FFFFF.csv
[works专用ip优选.zip](https://github.com/XIU2/CloudflareSpeedTest/files/13619207/works.ip.zip)

**Reply by blog2020** [reply]
麻烦大佬下载一下压缩包，谢谢。

**Reply by blog2020** [reply]
上传完我才发现压缩包里包括了CloudflareST.exe程序，难怪那么大，下载下面这个txt就可以了。
[ip.txt](https://github.com/XIU2/CloudflareSpeedTest/files/13619247/ip.txt)

**Reply by b1ghawk119** [reply]
学了
